dali/base/dafdesc.cpp

```diff
@@ -3653,6 +3654,36 @@ class CStoragePlaneInfo : public CInterfaceOf<IStoragePlane>
     virtual const char * querySingleHost() const override { return xml->queryProp("@host"); }   // MORE: Likely to be changed to resolve hosts
     virtual unsigned numDefaultSprayParts() const override { return xml->getPropInt("@defaultSprayParts", 1); }
     virtual bool queryDirPerPart() const override { return xml->getPropBool("@subDirPerFilePart", isContainerized()); } // default to dir. per part in containerized mode
+    virtual bool hasStorageApi() const override { return xml->hasProp("storageapi"); }
+    virtual StorageType getStorageType() const override
+    {
+        const char * storageTypeStr = xml->queryProp("storageapi/@type");
+        if (streq(storageTypeStr, "azurefile"))
+            return StorageType::StorageTypeAzureFile;
+        else if (streq(storageTypeStr, "azureblob"))
+            return StorageType::StorageTypeAzureBlob;
+        return StorageType::StorageTypeUnknown;
+    }
+    virtual const char * queryStorageApiAccount() const override
+    {
+        const char * storageApiAccount = xml->queryProp("storageapi/@account");
+        if (isEmptyString(storageApiAccount))
+            storageApiAccount = getenv("AZURE_ACCOUNT_NAME");
+        return storageApiAccount;
+    }
+    virtual const char * queryStorageContainer() const override
+    {
+        return xml->queryProp("storageapi/@container");
+    }
+    virtual StringBuffer & getSASToken(StringBuffer & token) const override
+    {
+        const char * secretName = xml->queryProp("storageapi/@secret");
+        if (isEmptyString(secretName))
+            return token;  // return empty string if no secret name is specified
+        getSecretValue(token, "storage", secretName, "token", false);
+        return token.trimRight();
```

Comment:
question: is trimRight necessary?

Reply:
I added this after trailing spaces was causing problems.

Reply:
right, because trimRight/clip only trims spaces, not ther whitespace chars..
Not a big deal, will work as is, but it begged the questions of why it was here, and won't acheive much by the sounds of it in case it might be useful.